tests/qchem/vibrational/test_pes_generator.py

```diff
@@ -0,0 +1,363 @@
+            ),
+        ),
+    ],
+)
+@pytest.mark.usefixtures("skip_if_no_pyscf_support", "skip_if_no_mpi4py_support")
+def test_onemode_pes(sym, geom, harmonic_res, do_dipole, exp_pes_onemode, exp_dip_onemode):
+    r"""Test that the correct onemode PES is obtained."""
+
+    mol = qml.qchem.Molecule(sym, geom, basis_name="6-31g", unit="Angstrom", load_data=True)
+    mol_eq = vibrational_class._single_point(mol)
+
+    gauss_grid, _ = np.polynomial.hermite.hermgauss(9)
+
+    freqs = harmonic_res["freq_wavenumber"]
+    displ_vecs = harmonic_res["norm_mode"]
+    pes_onebody, dipole_onebody = pes_generator._pes_onemode(
+        mol, mol_eq, freqs, displ_vecs, gauss_grid, method="RHF", dipole=do_dipole
+    )
+
+    assert np.allclose(pes_onebody, exp_pes_onemode, atol=1e-6)
+    if do_dipole:
+        assert np.allclose(dipole_onebody, exp_dip_onemode, atol=1e-6)
+    else:
+        assert dipole_onebody is None
+
+
+@pytest.mark.parametrize(
+    ("sym", "geom", "harmonic_res", "ref_file"),
+    # Expected results were obtained using vibrant code
+    [
+        (
+            ["H", "H", "S"],
+            np.array(
+                [
+                    [0.0, -1.00688408, -0.9679942],
+                    [0.0, 1.00688408, -0.9679942],
+                    [0.0, 0.0, -0.06401159],
+                ]
+            ),
+            {
+                "freq_wavenumber": np.array([1294.2195371, 2691.27147945, 2718.4023196]),
+                "norm_mode": np.array(
+                    [
+                        [
+                            [5.04812379e-17, -4.56823333e-01, 5.19946505e-01],
+                            [1.86137417e-16, 4.56823334e-01, 5.19946504e-01],
+                            [1.35223505e-17, -1.52311695e-11, -3.26953260e-02],
+                        ],
+                        [
+                            [-9.48723219e-18, -5.36044948e-01, -4.43104062e-01],
+                            [1.58760881e-16, 5.36044952e-01, -4.43104065e-01],
+                            [5.31102418e-17, -1.25299135e-10, 2.78633123e-02],
+                        ],
+                        [
+                            [6.52265536e-17, -5.15178992e-01, -4.62528763e-01],
+                            [3.12480546e-16, -5.15178988e-01, 4.62528760e-01],
+                            [1.63797627e-17, 3.23955347e-02, 9.23972875e-11],
+                        ],
+                    ]
+                ),
+            },
+            "H2S.hdf5",
+        )
+    ],
+)
+@pytest.mark.usefixtures("skip_if_no_pyscf_support", "skip_if_no_mpi4py_support")
+def test_twomode_pes(sym, geom, harmonic_res, ref_file):
+    r"""Test that the correct onemode PES is obtained."""
+
+    mol = qml.qchem.Molecule(sym, geom, basis_name="6-31g", unit="Angstrom", load_data=True)
+    mol_eq = vibrational_class._single_point(mol)
+
+    gauss_grid, _ = np.polynomial.hermite.hermgauss(9)
+
+    freqs = harmonic_res["freq_wavenumber"]
+    vectors = harmonic_res["norm_mode"]
```

Comment:
This is repeated in all three tests, wondering if it's possible to write a utility function to make it more concise? We can also avoid recomputing the onemode PES/twomode PES by saving it in a variable outside of the function scope.

Reply:
We are taking onemode_pes and twomode_pes from a file, I can read it outside the function scope.

Reply:
This doesn't work because of mpi, refactored the code a bit to make it more concise.